### PR TITLE
Simplify ssh box to fix multi-line command issues and add unit tests

### PR DESCRIPTION
**Pull request by xingyaoww**
On the current main branch, running multi-line command through sandbox:

```
# testing
print('test multi-line command')
exit_code, output = ssh_box.execute("pwd\nls -l\nls -l")
print('exit code:', exit_code)
print(output)
```

would timeout and/or have issues, since the while loop will be looking for `command_output` that startswith `cmd`.

image

This PR:
- remove those complex logic and fix this issue.
- add unit tests of single-line & multi-line commands to test

Will resolve https://github.com/OpenDevin/OpenDevin/issues/1431

### CODE REVIEW

File: tests/unit/test_sandbox.py
Comment (anthonyyoussef01) [inline tests/unit/test_sandbox.py:1]: Great job on simplifying the SSH box implementation!

File: tests/unit/test_sandbox.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        os.environ['RUN_AS_DEVIN'] = 'true'
+        os.environ['SANDBOX_TYPE'] = 'ssh'
+        from opendevin.sandbox.docker.ssh_box import DockerSSHBox
+        ssh_box = DockerSSHBox()
+
+        # test the ssh box
+        exit_code, output = ssh_box.execute('pwd\nls -l')
+        assert exit_code == 0, 'The exit code should be 0.'
+        print(output)
+        assert output.strip() == '/workspacels -l\r\ntotal 0'
```
Comment (anthonyyoussef01) [inline tests/unit/test_sandbox.py:56]: Could change this to 
```
expected_lines = ['/workspacels -l', 'total 0']
assert output.strip().splitlines() == expected_lines
```

File: tests/unit/test_sandbox.py
Comment (anthonyyoussef01) [reply tests/unit/test_sandbox.py:1]: I haven't pulled your changes, but if you could, please try adding the following test
```
def test_ssh_box_failed_cmd_run_as_devin():
    # get a temporary directory
    with tempfile.TemporaryDirectory() as temp_dir:
        pathlib.Path().mkdir(parents=True, exist_ok=True)
        os.environ['WORKSPACE_BASE'] = temp_dir
        os.environ['RUN_AS_DEVIN'] = 'true'
        os.environ['SANDBOX_TYPE'] = 'ssh'
        from opendevin.sandbox.docker.ssh_box import DockerSSHBox
        ssh_box = DockerSSHBox()

        # test the ssh box with a command that fails
        exit_code, output = ssh_box.execute('non_existing_command')
        assert exit_code != 0, 'The exit code should not be 0 for a failed command.'

    # cleanup the environment variables
    del os.environ['WORKSPACE_BASE']
    del os.environ['SANDBOX_TYPE']
```

Comment (xingyaoww) [reply tests/unit/test_sandbox.py:1]: Thanks for the suggestions! i add the test in the latest commit!

File: tests/unit/test_sandbox.py (hunk 1 above)
Comment (xingyaoww) [reply tests/unit/test_sandbox.py]: done!

**Review by rbren — APPROVED** [review]

**Comment by xingyaoww** [comment]
This PR is moved to https://github.com/OpenDevin/OpenDevin/pull/1460 since there's a bunch of issue with merge/rebase conflicts so i think it is easier & clenaer to start a new one

**Pull request closed without merging**